File: backend/app/schemas/resume.py
"""Resume schemas."""

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel
from app.core.constants import ResumeStatus


class ResumeResponse(BaseModel):
    id: UUID
    filename: str
    original_filename: str
    file_size_bytes: int | None
    mime_type: str | None
    status: ResumeStatus
    created_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/types/resume.ts
export type ResumeStatus = 'pending' | 'processing' | 'completed' | 'failed';

export interface ResumeResponse {
  id: string;
  filename: string;
  original_filename: string;
  file_size_bytes: number | null;
  mime_type: string | null;
  status: ResumeStatus;
  created_at: string;
}
